src/shave.cpp

```diff
@@ -296,14 +296,20 @@ class VTZeroGeometryTileFeature : public mbgl::GeometryTileFeature {
 };
 
 static bool evaluate(mbgl::style::Filter const& filter,
-                     float zoom,
+                     float minzoom,
+                     float maxzoom,
                      mbgl::FeatureType ftype,
                      vtzero::feature const& feature) // This properties arg is our custom type that we use in our lambda function below.
 {
     VTZeroGeometryTileFeature geomfeature(feature, ftype);
-    // std::string const& key is dynamic and comes from the Filter object
-    mbgl::style::expression::EvaluationContext context(zoom, &geomfeature);
-    return filter(context);
+
+    for(int zoom = floor(minzoom); zoom <= ceil(maxzoom); zoom++) {
```

Comment:
Please use `std::floor` and `std::ceil` to ensure we're using the C++ standard library here rather than the C lib.